### PR TITLE
Use `argparse` for Managing REST Arguments [minor]

### LINKED ISSUE #97: Investigate Extending `argparse` for `get_argument()`
`get_argument()` has become a lot larger than I originally planned. In an effort to avoid bugs and limit scope creep, using the `argparse` package may be beneficial.

### PR DESCRIPTION
**Pull request by ric-evans**
Advanced server-side REST argument handling is done using `ArgumentHandler`'s `argparse`-encapsulated methods. See README: https://github.com/WIPACrepo/rest-tools/blob/89d31efebd6e6cf0543a8571a7260f47afde4f0f/README.md#handling-arguments-server-side

Deprecated:
- `RestHandler.get_json_body_argument()`
- `RestHandler.get_argument()`'s advanced argument checking parameters
   + `type`
   + `choices`
   + `forbiddens`
   + `strict_type`
- `RestHandler.get_argument()`'s return type is no longer typehinted (previously used `TypeVar`)

Backward Compatibility:
- `RestHandler.get_argument()` will continue to look at the JSON-decoded request body (without the above-deprecated features)
	+ optimized JSON decoding

Also, adds Python 3.12 support

closes #97

### CODE REVIEW

**Comment by ric-evans** [comment]
@dsschult `test_openid_login_handler_get()` is failing, could you look into this? I don't believe anything I've touched could have affected this

File: rest_tools/server/handler.py [hunk 1]
```diff
@@ -171,6 +171,19 @@ def prepare(self):
...
+            try:
+                self.request.body_arguments = json_decode(self.request.body)
+            except json.JSONDecodeError:
+                raise tornado.web.HTTPError(
+                    400, reason="requests body is not JSON-encoded"
+                )
+            if not isinstance(self.request.body_arguments, dict):
+                raise tornado.web.HTTPError(
+                    400, reason="JSON-encoded requests body must be a 'dict'"
+                )
```
Comment (dsschult) [inline rest_tools/server/handler.py:185]: I would prefer not to do this in `prepare()` as that applies to all requests everywhere.  Maybe use a cached function in `arghandler.py`, so it's only ever done once?

Comment (ric-evans) [reply rest_tools/server/handler.py]: [`4491d41` (#138)](https://github.com/WIPACrepo/rest-tools/pull/138/commits/4491d4135b77be3adb0c8501cb240b88ea901ab2)

File: tests/unit_server/rest_handler_test.py [hunk 2]
```diff
@@ -271,13 +283,41 @@ async def fn2(*args, **Kwargs):
 
     handler.get_authenticated_user = MagicMock(side_effect=fn2)
 
-    request.body = '{"code": "thecode", "state": "state"}'
+    request.body = '{"code": "thecode", "state": "state"}'  # invalid state
```
Comment (dsschult) [inline tests/unit_server/rest_handler_test.py:286]: Can you explain this more?  What's invalid about this state?  It won't pass `_decode_state()`, but that's mocked out.  Technically it's legal according to the oauth2 spec.

Comment (ric-evans) [reply tests/unit_server/rest_handler_test.py:286]: I missed that mocking. I figured that `_decode_state()` was raising the `HTTPError` and triggered `handler.write()`

Comment (ric-evans) [reply tests/unit_server/rest_handler_test.py:286]: it looks that `get_authenticated_user()` raises the error, yeah?

Comment (dsschult) [reply tests/unit_server/rest_handler_test.py:286]: That should also be mocked out?

Is the test diff just pulling each case apart into separate functions, and there's no actual test changes?

Comment (ric-evans) [reply tests/unit_server/rest_handler_test.py]: ah I see now https://github.com/WIPACrepo/rest-tools/blob/argparse-args/rest_tools/server/handler.py#L460

**Pull request merged**